nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -69,6 +98,41 @@ void LatticeMotionTable::initMotionModel(
 
   // TODO(Matt) populate num_angle_quantization, size_x, min_turning_radius, trig_values,
   // all of the member variables of LatticeMotionTable
+
+  num_angle_quantization = latticeMetadata.number_of_headings;
+  num_angle_quantization_float = static_cast<float>(num_angle_quantization);
+  min_turning_radius = latticeMetadata.min_turning_radius;
+
+  //Find the largest turning radius that is not equal to zero
+  float largestTurningRadius = std::numeric_limits<float>::max();
+  for(auto const& motionPrimitivesAtAngle : motionPrimitives)
+  {
+    for(auto const& motionPrimitive : motionPrimitivesAtAngle)
+    {
+      if( motionPrimitive.radius < largestTurningRadius && motionPrimitive.radius != 0 )
+      {
+        largestTurningRadius = motionPrimitive.radius;
+      }
+    }
+  }
+
+  //NOTE: Note sure what type of state space this should be 
+  state_space = std::make_unique<ompl::base::DubinsStateSpace>( largestTurningRadius );
```

Comment:
You don't need to find the `largestTurningRadius`, you should use what you already set above for this instead, `min_turning_radius` :wink: that should let you remove the above double nested for loop. The goal is to find the smallest allowable circular radius it can turn in, not the largest (since the "largest" can be infinitely large until its essentially a straight line)

Reply:
Done